Importer/Log/CsvLog.php

```diff
@@ -50,9 +50,12 @@ public function addDataError($index, $errorMessage)
     /**
      * {@inheritdoc}
      */
-    public function addCriticalError($errorMessage)
+    public function addCriticalError($message, $userError = false)
```

Comment:
Well, it's an application/critical error, you should **always** have something to log as a `errorReason`.
It's the user message (`userMessage`) that is optional. Sometimes we can give a user a hint what went wrong, e.g. "no many errors found, it doesn't make sense to continue".

It's definitely not perfect, but for it somehow is a tiny bit better than having a flag parameter at the end.